### PR TITLE
feat(metadata-views): Save inline edited metadata

### PR DESCRIPTION
**Pull request by manishPh**

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @manishPh has signed the CLA. Thanks for the pull request!

File: src/features/metadata-based-view/__tests__/MetadataBasedItemList.test.js [hunk 1]
```diff
@@ -86,13 +86,35 @@ describe('features/metadata-based-view/MetadataBasedItemList', () => {
...
+
     const getWrapper = (props = defaultProps) => mount(<MetadataBasedItemList {...props} />);
 
     beforeEach(() => {
         wrapper = getWrapper();
         instance = wrapper.instance();
     });
 
+    describe('componentDidUpdate()', () => {
+        test('should call setState when component gets updated with different props', () => {
```
Comment (sahiga) [inline src/features/metadata-based-view/__tests__/MetadataBasedItemList.test.js]: Should we add another test to ensure that `setState` is not called when the props haven't changed, i.e., `isEqual(currentItems, prevItems) === true`?

File: src/features/metadata-based-view/MetadataBasedItemList.js [hunk 2]
```diff
@@ -95,23 +98,44 @@ class MetadataBasedItemList extends React.Component<Props, State> {
...
-        const itemWithPreviewPermission = { ...item, permissions };
+        const permissions = {
+            can_preview: true,
+            can_upload: true,
+        };
+        return { ...item, permissions };
+    };
 
-        onItemClick(itemWithPreviewPermission);
+    getMetadataColumnName(column: MetadataColumnConfig | string): string {
```
Comment (sahiga) [inline src/features/metadata-based-view/MetadataBasedItemList.js]: Can we add tests for `getMetadataColumnName`, `handleEditIconClick`, and `handleSave`?

Comment (manishPh) [reply src/features/metadata-based-view/MetadataBasedItemList.js]: Done

File: src/features/metadata-based-view/__tests__/MetadataBasedItemList.test.js (hunk 1 above)
Comment (manishPh) [reply src/features/metadata-based-view/__tests__/MetadataBasedItemList.test.js]: Done

**Pull request merged**